src/Features/CSharp/Portable/UseSimpleUsingStatement/UseSimpleUsingStatementCodeFixProvider.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseSimpleUsingStatement
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(UseSimpleUsingStatementCodeFixProvider)), Shared]
+    internal class UseSimpleUsingStatementCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseSimpleUsingStatementDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, context.Diagnostics[0], c)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var topmostUsingStatements = diagnostics.Select(d => (UsingStatementSyntax)d.AdditionalLocations[0].FindNode(cancellationToken)).ToSet();
+            var blocks = topmostUsingStatements.Select(u => (BlockSyntax)u.Parent);
+
+            var root = editor.OriginalRoot;
+            var updatedRoot = root.ReplaceNodes(
+                blocks,
+                (original, current) => RewriteBlock(original, current, topmostUsingStatements));
+
+            editor.ReplaceNode(root, updatedRoot);
+
+            return Task.CompletedTask;
+        }
+
+        private static SyntaxNode RewriteBlock(
+            BlockSyntax originalBlock, BlockSyntax currentBlock,
+            ISet<UsingStatementSyntax> topmostUsingStatements)
+        {
+            if (originalBlock.Statements.Count == currentBlock.Statements.Count)
+            {
+                var statementToUpdateIndex = originalBlock.Statements.IndexOf(s => topmostUsingStatements.Contains(s));
```

Comment:
You're doing `IndexOf` then getting the statement at that index. Maybe `FirstOrDefault`?

Reply:
i'm getting the statement at that index in the *current list of children*.  I need the index in the *original* list.   Since the lists are different, i can't use FirstOrDefault.

#bydesign.